hikari/api/chunker.py

```diff
@@ -28,8 +28,61 @@
 import abc
 import typing
 
+from hikari import undefined
+
 if typing.TYPE_CHECKING:
     from hikari import guilds
+    from hikari import iterators
+    from hikari import snowflakes
+    from hikari import users as users_
+    from hikari.events import shard_events
+
+
+class ChunkInformation(typing.Protocol):
+    """Information about a member request that's being tracked."""
+
+    @property
+    def average_chunk_size(self) -> typing.Optional[int]:
+        """Average amount of members that are being received per chunk."""
+
+    @property
+    def chunk_count(self) -> typing.Optional[int]:
+        """Amount of chunks that are expected for this request."""
+
+    @property
+    def guild_id(self) -> snowflakes.Snowflake:
+        """Snowflake ID of the guild this chunk request is for."""
+
+    @property
+    def is_complete(self) -> bool:
+        """Whether this chunk event is finished or not.
+
+        A chunk event may be considered finished after all chunks have been received
+        or after it's timed out.
+        """
+
+    @property
+    def last_received(self) -> typing.Optional[int]:
+        """Monotonic time of when we last received a chunk for this event in ms."""
```

Comment:
oh that's a typo, should be ns not ms

Reply:
Tbf i was thinking about changing that to a date time and just storing the monotonic time along side a datetime just because datetimes are more user friendly